internal/statestore/backfill_test.go

```diff
@@ -604,3 +604,57 @@ func generateBackfills(ctx context.Context, t *testing.T, service Service, amoun
 
 	return backfills
 }
+
+func TestCleanupBackfills(t *testing.T) {
+	cfg, closer := createRedis(t, false, "")
+	defer closer()
+	service := New(cfg)
+	require.NotNil(t, service)
+	defer service.Close()
+	ctx := utilTesting.NewContext(t)
+
+	rc, err := redis.Dial("tcp", fmt.Sprintf("%s:%s", cfg.GetString("redis.hostname"), cfg.GetString("redis.port")))
+	require.NoError(t, err)
+
+	bfID := "mockBackfill-1"
+	ticket1ID := "t1"
+	ticket2ID := "t2"
+	ticketIDs := []string{ticket1ID, ticket2ID}
+	bfLastAck := "backfill_last_ack_time"
+	proposedTicketIDs := "proposed_ticket_ids"
+
+	// ARRANGE
+	err = service.CreateBackfill(ctx, &pb.Backfill{
+		Id:         bfID,
+		Generation: 1,
+	}, ticketIDs)
+	require.NoError(t, err)
+
+	// add expired but acknowledged backfill
+	_, err = rc.Do("ZADD", bfLastAck, 123, bfID)
+	require.NoError(t, err)
+
+	err = service.AddTicketsToPendingRelease(ctx, ticketIDs)
+	require.NoError(t, err)
+
+	// ACT
+	err = service.CleanupBackfills(ctx)
+	require.NoError(t, err)
+
+	// ASSERT
+
+	// backfill doesn't exist anymore
+	_, _, err = service.GetBackfill(ctx, bfID)
+	require.Error(t, err)
+	require.Equal(t, "Backfill id: mockBackfill-1 not found", status.Convert(err).Message())
```

Comment:
simply checking error message equality is sufficient? it would fail anyways if there is no error.

Reply:
Checking for an error explicitly provides more precise message.
Example:
`Error:      	An error is expected but got nil.`
I clearly see that there should be an error but the function returned nil.

If I leave only comparing error messages I will get:
```        	
Error:      	Not equal: 
expected: "Backfill id: mockBackfill-1 not found"
actual  : ""
```
And here I'm not sure - is error nil or is its description just empty for some reason?

Reply:
In your sample output, the the actual error being "" is clearly indicative of the error returned being empty right? Still not sure what extra value the extra check adds. The test expects an error - and the returned error is not that - and thats the failure. Unless this error does not catch some output, I do not see the value for the extra check.

Reply:
Message `actual  : ""` doesn't give an information where it was nil error returned or actual error with an empty string description. `status.Convert(err).Message()` converts nil error to an empty string.

Reply:
Does it matter if it returned nil - or "" - both of which are unexpected when the expected value is "Backfill id: mockBackfill-1 not found"? What debugging value is added by understanding that it returned nil - over not being sure if it was nil or empty string? I wont block subject my approval over this - but am not convinced that the added check adds any value to the test.